base/src/commonMain/kotlin/monitor/ServiceMonitorState.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.monitor
+
+import com.splendo.kaluga.base.ServiceMonitor
+import com.splendo.kaluga.base.flow.SpecialFlowValue
+import com.splendo.kaluga.state.ColdStateFlowRepo
+import com.splendo.kaluga.state.State
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+sealed class ServiceMonitorState : State() {
+
+    sealed class Initialized(
+        private val monitor: ServiceMonitor
+    ) : ServiceMonitorState() {
+        val notInitialized: suspend () -> NotInitialized = { NotInitialized(monitor) }
+
+        fun deinitialize(): suspend () -> NotInitialized {
+            monitor.stopMonitoring()
+            return notInitialized
+        }
+
+        class Enabled(monitor: ServiceMonitor) : Initialized(monitor) {
+            val disabled: suspend () -> Disabled = { Disabled(monitor) }
+        }
+        class Disabled(monitor: ServiceMonitor) : Initialized(monitor) {
+            val enabled: suspend () -> Enabled = { Enabled(monitor) }
+        }
+    }
+
+    class NotInitialized(
+        private val monitor: ServiceMonitor
+    ) : ServiceMonitorState(), SpecialFlowValue.NotImportant {
+        val initialized: suspend () -> Initialized = { Initialized.Enabled(monitor) }
+        val initializedDisabled: suspend () -> Initialized = { Initialized.Disabled(monitor) }
+
+        fun initialize(): suspend () -> Initialized {
+            monitor.startMonitoring()
+            return if (monitor.isServiceEnabled) {
+                initialized
+            } else {
+                initializedDisabled
+            }
+        }
+    }
+}
+
+class ServiceMonitorStateRepo(
+    monitor: ServiceMonitor,
```

Comment:
Important thing here, we don't need both a `ServiceMonitor` and a `ServiceMonitorStateRepo`, one StateRepo based API is enough.

Reply:
I think keeping them separated makes things easier, both for testing and usage. if you want to avoid user to use `ServiceMonitor` I could make it internal so that they have no access to it